### ISSUE TITLE
"Mixing and Matching" wiki example doesn't compile

### ISSUE THREAD

**Issue by elexisvenator** [jonwagner/Insight.Database#189]
I have a stored procedure that returns a list of Competency followed by a one-to-one mapped list of FieldOfEducation, and then the same again (so result is 2 lists of competencies with matching lists of FieldOfEducation). This looks quite similar to your [Mixing and Matching](https://github.com/jonwagner/Insight.Database/wiki/Specifying-Result-Structures#mixing-and-matching) example. I wrote the query as below.

```C#
Results<Competency, Competency>  results = c.Query(
                "dbo.getCompetenciesForId",
                params, 
                returns: Query.Returns(Some<Competency>.Records)
                     .ThenChildren(Some<FieldOfEducation>.Records)
                     .Then(Some<Competency>.Records)
                     .ThenChildren(Some<FieldOfEducation>.Records));
```

This is almost identical to the example in the wiki (the example contains a semicolon in the wrong place and missing brackets). This however does not work as there is no overload for `.ThenChildren` that accepts `ResultsReader<T1,T2>`. I'm not sure how the example in the wiki is meant to work, or if it is showing a different example to what I am attempting.

Anyone got any advice?

Other notes:
.Net4.5
Insight Database v5.2.1 (Nuget)
Using MSSQL Server as provider

**Comment by jonwagner** [comment]
That is supposed to work. I'll have to dig into it. You're probably right that there is a missing extension method.

**Comment by jonwagner** [comment]
I added the extension methods. I can't check the code in from here, but I'll do a build tomorrow and push out a new version.

**Comment by jonwagner** [comment]
This fix is in v5.2.2, in nuget.
